src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -842,8 +842,18 @@ def _build_preflight_error_message(preflight_error):
     return '\n'.join(err_messages)
 
 
+def _prepare_template_uri_with_query_string(template_uri, input_query_string):
+    from six.moves.urllib.parse import urlencode, parse_qs, urlsplit, urlunsplit  # pylint: disable=import-error
+
+    scheme, netloc, path, query_string, fragment = urlsplit(template_uri)  # pylint: disable=unused-variable
+    query_params = parse_qs(input_query_string)
+    new_query_string = urlencode(query_params, doseq=True)
+
+    return urlunsplit((scheme, netloc, path, new_query_string, fragment))
```

Comment:
Do we need to consider adding some verifications for `input_query_string` to make sure it is vaild?